google-cloud-datastore/src/main/java/com/google/cloud/datastore/spi/DefaultDatastoreRpc.java

```diff
@@ -130,7 +134,7 @@ public CommitResponse commit(CommitRequest request) {
     try {
       return client.commit(request);
     } catch (com.google.datastore.v1.client.DatastoreException ex) {
-      throw translate(ex);
+      throw translate(ex, request.getMode() == CommitRequest.Mode.NON_TRANSACTIONAL);
```

Comment:
We could add a retry-logic to [`runInTransaction`](https://github.com/GoogleCloudPlatform/google-cloud-java/blob/master/google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreHelper.java#L92) as a feature later on.